src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -49,6 +52,28 @@ public class ArduinoBoard : Board.Board, IDisposable
 
         private ILogger _logger;
 
+        static ArduinoBoard()
+        {
+            // We add all known modes to the list, even though we don't really support them all in the core
+            _knownSupportedModes.Add(SupportedMode.DigitalInput);
```

Comment:
I'd recommend to do the initialization a bit differently:
start with `s_knownSupportedModes` being null, then create a `List<SupportedMode>` local, add all values then assign it to `s_knownSupportedModes`, perhaps even create a private helper method like `GetKnownSupportedValues` and just call it from field initializer. It's very easy to accidentally end up in semi-initialized state when doing this, with local/method you are either null or fully initialized and it's much easier to debug, reason or refactor this code